### PR TITLE
Prevent uploading `.git` folder

### LINKED ISSUE #238: Uploading folder with .git causes problems

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
Reject all the files under the `.git` folder when uploading.
##
Fixes #238

### CODE REVIEW

File: frontend/src/components/DropZone.js [hunk 1]
```diff
@@ -37,6 +40,24 @@ const activeStyle = {
...
+ * @returns {FileError|null}
+ */
+const fileCustomValidator = file => {
+  const isFileUnderGitDir =
+    file?.path.includes('/.git/') || file?.webkitRelativePath.includes('/.git/')
+
+  if (isFileUnderGitDir) {
+    return {
+      code: ErrorCode.FileInvalidType,
+      message: "Uploading git folder isn't supported",
```
Comment (kasbah) [inline frontend/src/components/DropZone.js:55]: I would prefer to just ignore these files.

Comment (AbdulrhmnGhanem) [reply frontend/src/components/DropZone.js:55]: Now it ignores it and uploads the rest but it shows the message shown below, do you mean ignoring it silently?

image

Comment (AbdulrhmnGhanem) [reply frontend/src/components/DropZone.js:55]: It has been deployed to https://abdo-dev.staging.kitspace.dev/

Comment (kasbah) [reply frontend/src/components/DropZone.js:55]: That warning disappears so quickly it seems useless anyway. I think it's fine to just ignore these files silently.

Comment (AbdulrhmnGhanem) [reply frontend/src/components/DropZone.js:57]: Resolved in 56f9c4f.

**Pull request merged**